### PR TITLE
Added slug column to the CalendarEvent

### PR DESCRIPTION
**Pull request by coderrsid**
## Issue
fixes: #78

### CODE REVIEW

**Comment by coderrsid** [comment]
Hey @hereisnaman.. i am able to generate slug perfectly.. As you can see below API side, 

image

but the problem i am getting is.. although i have done no changes to the calendarEventCreate Mutation. Is it because the `deletedAt` value is null? 
image

**Comment by 0xnmn** [comment]
> Hey @hereisnaman.. i am able to generate slug perfectly.. As you can see below API side,
> 
> image
> 
> but the problem i am getting is.. although i have done no changes to the calendarEventCreate Mutation. Is it because the `deletedAt` value is null?
> image

Can you please share the server logs. Also try re-basing with master.

File: api/services/AuthorizationPolicy/index.js [hunk 1]
```diff
@@ -108,6 +109,10 @@ policy.include('calendarEvent', (p) => {
         isOrganiser(event, viewer) || isMember(viewer),
     );
   });
+
+  p.include('slug', (cp) => {
```
Comment (0xnmn) [inline api/services/AuthorizationPolicy/index.js:113]: not required.

File: api/services/CalendarEvent/index.js [hunk 2]
```diff
@@ -4,10 +4,13 @@ import BaseModelService, {
...
   withTransaction,
 } from 'Services/BaseModelService';
+import slugify from 'slugify';
 
 export default class CalendarEvent extends BaseModelService {
   @saveInstance
   create(body) {
+    let generatedSlug = slugify(body.title);
+    body.slug = generatedSlug;
     return Models.CalendarEvent.create(body).then((event) =>
```
Comment (0xnmn) [inline api/services/CalendarEvent/index.js:14]:
```javascript
.create({
  slug: slugify(body.title),
  ...body,
})
```

File: app/components/Events/Form.js [hunk 3]
```diff
@@ -56,6 +58,17 @@ const EventForm = memo(
...
+          <Form.Group className="has-float-label mb-4 col-12">
+            <Form.Label>Slug</Form.Label>
+            <Form.Input
+              type="text"
+              name="slug"
+              value={slug}
+              onChange={(e) => setSlug(e.target.value)}
+            />
+          </Form.Group>
+        )}
```
Comment (0xnmn) [inline app/components/Events/Form.js:71]: Should me rendered always for edit show. Have an extra prop `showSlug` and pass it as true from EditModal

File: app/modals/ViewEvent/Content/index.js [hunk 4]
```diff
@@ -64,6 +64,11 @@ const EventContent = memo(({ event, onClose }) => {
   return (
     <>
       <ModalHeader className="w-100">
+        <div className="d-flex">
+          <span className="w-50 text-small">
+            <a href={`event/${event.slug}`}>events/{event.slug}</a>
```
Comment (0xnmn) [inline app/modals/ViewEvent/Content/index.js:69]: `/events/{event.slug}`

Also add screenshots.

**Review by 0xnmn — CHANGES_REQUESTED** [review]

**Comment by coderrsid** [comment]
Please visit https://github.com/coding-blocks/gondor/pull/102 for further changes.

**Pull request closed without merging**